zk/stages/stage_batches.go

```diff
@@ -686,7 +687,8 @@ func UnwindBatchesStage(u *stagedsync.UnwindState, tx kv.RwTx, cfg BatchesCfg, c
 	// store the highest used l1 info index//
 	/////////////////////////////////////////
 
-	highestL1InfoTreeIndex, err := hermezDb.GetLatestL1InfoTreeIndex()
+	// highestL1InfoTreeIndex, err := hermezDb.GetLatestL1InfoTreeIndex()
+	highestL1InfoTreeIndex, err := hermezDb.GetBlockL1InfoTreeIndexProgress(fromBlock)
```

Comment:
Because we don't use an index for each block I think we'll need another way of handling this, we should navigate backwards from the `fromBlock` until we find one that has been used.  Maybe the mdbx cursor can help here?

Reply:
If we're storing changes to the highest used l1 info tree index in a table then I'd argue we can get rid of any code that is using the stages.HighestUsedL1InfoIndex completely in favour of this which we can unwind.

Reply:
So your idea is to keep only blocks where there is an actual change in the progress? If yes, I think we can apply the same logic to the L1InfoTreeIndex itself. What do you think?

I totally agree that we can remove the stage progress in favor of the DB table.

Reply:
That would be my idea yup, just to keep relevant records when things change.  It is for storage usage yes, and to make things clear in the DB if we ever need to take a look.  It would have been nice to just have a single table holding this information but as we found with bad batches we need to keep the highest used index as a separate thing to detect this.

Reply:
I've made changes. Please have a look.